File: app/src/main/java/com/pcs/ztqtj/view/activity/service/ActivityServerHyqx.java
package com.pcs.ztqtj.view.activity.service;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.BaseAdapter;
import android.widget.GridView;
import android.widget.ImageView;
import android.widget.TextView;

import com.pcs.lib.lib_pcs_v3.model.image.ImageConstant;
import com.pcs.lib_ztqfj_v2.model.pack.net.column.ColumnInfo;
import com.pcs.lib_ztqfj_v2.model.pack.net.column.PackColumnDown;
import com.pcs.ztqtj.MyApplication;
import com.pcs.ztqtj.R;
import com.pcs.ztqtj.control.tool.utils.TextUtil;
import com.pcs.ztqtj.util.CONST;
import com.pcs.ztqtj.util.OkHttpUtil;
import com.pcs.ztqtj.view.activity.FragmentActivityZtqBase;
import com.pcs.ztqtj.view.activity.product.agriculture.ActivityAgricultureWeatherColumn;
import com.pcs.ztqtj.view.activity.product.traffic.ActivityTraffic;

import org.jetbrains.annotations.NotNull;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

import okhttp3.Call;
import okhttp3.Callback;
import okhttp3.FormBody;
import okhttp3.MediaType;
import okhttp3.Request;
import okhttp3.RequestBody;
import okhttp3.Response;

/**
 * 专项服务-行业气象
 */
public class ActivityServerHyqx extends FragmentActivityZtqBase {

    private MyGridViewAdapter mGridViewAdapter;

    @Override
    protected void onCreate(Bundle arg0) {
        super.onCreate(arg0);
        setContentView(R.layout.activity_server_hyqx);
        setTitleText("行业气象");
        createImageFetcher();
        initGridView();
        okHttpColumnList();
    }

    /**
     * 初始化GridView
     */
    private void initGridView() {
        GridView mGridView = findViewById(R.id.gridview);
        mGridViewAdapter = new MyGridViewAdapter(this);
        mGridView.setAdapter(mGridViewAdapter);
    }

    /**
     * 刷新GridView
     */
    private void refreshGridView() {
        mGridViewAdapter.notifyDataSetChanged();
    }

    /**
     * GridView适配器
     * @author JiangZy
     */
    private class MyGridViewAdapter extends BaseAdapter {

        private Context mContext;

        public MyGridViewAdapter(Context context) {
            mContext = context;
        }

        @Override
        public int getCount() {
            if (arrcolumnInfo == null) {
                return 0;
            }
            return arrcolumnInfo.size();
        }

        @Override
        public Object getItem(int position) {
            if (arrcolumnInfo == null) {
                return null;
            }
            return position;
        }

        @Override
        public long getItemId(int position) {
            if (arrcolumnInfo == null) {
                return 0;
            }
            return position;
        }

        @Override
        public View getView(final int position, View view, ViewGroup parent) {
            Holder holder;
            if (view == null) {
                holder = new Holder();
                view = LayoutInflater.from(mContext).inflate(R.layout.item_server_second, null);
                holder.itemImageView = (ImageView) view.findViewById(R.id.itemImageView);
                holder.itemimageview_top = (ImageView) view.findViewById(R.id.itemimageview_top);
                holder.itemName = (TextView) view.findViewById(R.id.itemName);
                view.setTag(holder);
            } else {
                holder = (Holder) view.getTag();
            }

            holder.itemimageview_top.setVisibility(View.GONE);
            final ColumnInfo info = arrcolumnInfo.get(position);
            if (!TextUtils.isEmpty(info.ioc)) {
                String url = mContext.getString(R.string.file_download_url) + info.ioc;
                getImageFetcher().loadImage(url, holder.itemImageView, ImageConstant.ImageShowType.SRC);
            }

            holder.itemImageView.setOnClickListener(new View.OnClickListener() {

                @Override
                public void onClick(View arg0) {
                    Intent intent = null;
                    if (info.type.equals("10103030201")) {
                        intent = new Intent(mContext, ActivityTraffic.class);
                    } else if (info.type.equals("10103030202")) {
                        intent = new Intent(mContext, ActivityAgricultureWeatherColumn.class);
                    }
                    if (intent != null) {
                        intent.putExtra("type", info.type);
                        startActivity(intent);
                    }
                }
            });
            holder.itemName.setText(info.name);
            return view;
        }

        private class Holder {
            public ImageView itemImageView;
            public ImageView itemimageview_top;
            public TextView itemName;
        }
    }

    public List<ColumnInfo> arrcolumnInfo = new ArrayList<>();

    /**
     * 获取行业气象列表
     */
    private void okHttpColumnList() {
        showProgressDialog();
        new Thread(new Runnable() {
            @Override
            public void run() {
                try {
                    JSONObject param  = new JSONObject();
                    param.put("token", MyApplication.TOKEN);
                    String json = param.toString();
                    final String url = CONST.BASE_URL+"column_list";
                    Log.e("column_list", url);
                    RequestBody body = FormBody.create(MediaType.parse("application/json; charset=utf-8"), json);
                    OkHttpUtil.enqueue(new Request.Builder().post(body).url(url).build(), new Callback() {
                        @Override
                        public void onFailure(@NotNull Call call, @NotNull IOException e) {
                        }
                        @Override
                        public void onResponse(@NotNull Call call, @NotNull Response response) throws IOException {
                            if (!response.isSuccessful()) {
                                return;
                            }
                            final String result = response.body().string();
                            runOnUiThread(new Runnable() {
                                @Override
                                public void run() {
                                    Log.e("column_list", result);
                                    if (!TextUtil.isEmpty(result)) {
                                        try {
                                            JSONObject obj = new JSONObject(result);
                                            if (!obj.isNull("b")) {
                                                JSONObject bobj = obj.getJSONObject("b");
                                                if (!bobj.isNull("column")) {
                                                    JSONObject yjxx_info_query = bobj.getJSONObject("column");
                                                    if (!TextUtil.isEmpty(yjxx_info_query.toString())) {
                                                        dismissProgressDialog();
                                                        PackColumnDown down = new PackColumnDown();
                                                        down.fillData(yjxx_info_query.toString());
                                                        arrcolumnInfo.clear();
                                                        arrcolumnInfo.addAll(down.arrcolumnInfo);
                                                        refreshGridView();
                                                    }
                                                }
                                            }
                                        } catch (JSONException e) {
                                            e.printStackTrace();
                                        }
                                    }
                                }
                            });
                        }
                    });
                } catch (JSONException e) {
                    e.printStackTrace();
                }
            }
        }).start();
    }

}

File: app/src/main/java/com/pcs/ztqtj/view/activity/service/ActivityServerHyqx.kt
package com.pcs.ztqtj.view.activity.service

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import android.widget.TextView
import com.pcs.ztqtj.R
import com.pcs.ztqtj.model.ZtqCityDB
import com.pcs.ztqtj.util.CONST
import com.pcs.ztqtj.util.ColumnDto
import com.pcs.ztqtj.view.activity.FragmentActivityZtqBase
import com.pcs.ztqtj.view.activity.agricuture.ActivityAgricutureFact
import com.pcs.ztqtj.view.activity.agricuture.ActivityAgricutureView
import com.pcs.ztqtj.view.activity.photoshow.ActivityLogin
import com.pcs.ztqtj.view.activity.photoshow.ActivityPhotoShow
import com.pcs.ztqtj.view.activity.product.agriculture.ActivityAgricultureWeather
import com.pcs.ztqtj.view.activity.product.traffic.ActivityTraffic
import com.pcs.ztqtj.view.activity.prove.AgriEachActivity
import com.pcs.ztqtj.view.activity.web.webview.ActivityWebView
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.activity_server_hyqx.*

/**
 * 专项服务-行业气象
 */
class ActivityServerHyqx : FragmentActivityZtqBase() {

    private val dataList: ArrayList<ColumnDto> = ArrayList()
    private var mAdapter: MyGridViewAdapter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_server_hyqx)
        initGridView()
    }

    private fun initGridView() {
        if (intent.hasExtra("title")) {
            val title = intent.getStringExtra("title")
            if (title != null) {
                titleText = title
            }
        }

        if (intent.hasExtra("dataList")) {
            dataList.addAll(intent.getParcelableArrayListExtra("dataList"))
        }
        mAdapter = MyGridViewAdapter(this)
        gridview.adapter = mAdapter
    }

    private inner class MyGridViewAdapter(private val mContext: Context) : BaseAdapter() {

        private inner class Holder {
            var itemImageView: ImageView? = null
            var itemimageview_top: ImageView? = null
            var itemName: TextView? = null
        }

        override fun getCount(): Int {
            return dataList.size
        }

        override fun getItem(position: Int): Any {
            return position
        }

        override fun getItemId(position: Int): Long {
            return position.toLong()
        }

        override fun getView(position: Int, itemView: View?, parent: ViewGroup): View {
            var view = itemView
            val holder: Holder
            if (view == null) {
                holder = Holder()
                view = LayoutInflater.from(mContext).inflate(R.layout.adapter_special_service, null)
                holder.itemImageView = view.findViewById<View>(R.id.itemImageView) as ImageView
                holder.itemimageview_top = view.findViewById<View>(R.id.itemimageview_top) as ImageView
                holder.itemName = view.findViewById<View>(R.id.itemName) as TextView
                view.tag = holder
            } else {
                holder = view.tag as Holder
            }

            val data: ColumnDto = dataList[position]

            if (!TextUtils.isEmpty(data.icon)) {
                val url = mContext.getString(R.string.msyb) + data.icon
                Picasso.get().load(url).error(R.drawable.no_pic).into(holder.itemImageView)
            } else {
                holder.itemImageView!!.setImageResource(R.drawable.no_pic)
            }
            holder.itemImageView!!.setOnClickListener {
                when(data.dataCode) {
                    "10103030201" -> {//交通气象
                        val intent = Intent(mContext, ActivityTraffic::class.java)
                        intent.putExtra("title", data.dataName)
                        intent.putExtra("dataCode", data.dataCode)
                        startActivity(intent)
                    }
                    "101030302020101" -> {//农情实况数据
                        val intent = Intent(mContext, ActivityAgricutureFact::class.java)
                        intent.putExtra("title", data.dataName)
                        intent.putExtra("dataCode", data.dataCode)
                        startActivity(intent)
                    }
                    "101030302020102" -> {//农田实景
                        val intent = Intent(mContext, ActivityAgricutureView::class.java)
                        intent.putExtra("title", data.dataName)
                        intent.putExtra("dataCode", data.dataCode)
                        startActivity(intent)
                    }
                    "101030302020201" -> {//旬报
                        val intent = Intent(mContext, ActivityAgricultureWeather::class.java)
                        intent.putExtra("title", data.dataName)
                        intent.putExtra("dataCode", data.dataCode)
                        intent.putExtra("channel_id", "10103030202")
                        intent.putExtra("type", "xb")
                        startActivity(intent)
                    }
                    "101030302020202" -> {//月报
                        val intent = Intent(mContext, ActivityAgricultureWeather::class.java)
                        intent.putExtra("title", data.dataName)
                        intent.putExtra("dataCode", data.dataCode)
                        intent.putExtra("channel_id", "10103030202")
                        intent.putExtra("type", "yb")
                        startActivity(intent)
                    }
                    "101030302020301" -> {//农情苗情速报
                        if (!ZtqCityDB.getInstance().isLoginService()) {
                            intent = Intent(mContext, ActivityLogin::class.java)
                            startActivityForResult(intent, CONST.RESULT_LOGIN)
                        } else {
                            val intent = Intent(mContext, ActivityPhotoShow::class.java)
                            intent.putExtra("title", data.dataName)
                            intent.putExtra("imgType", "2")//imgType:图片类型，1（实景开拍），2（农业开拍分类）必须传，区分哪个业务
                            startActivity(intent)
                        }
                    }
                    "101030302020302" -> {//农情苗情互动
                        val intent = Intent(mContext, AgriEachActivity::class.java)
                        intent.putExtra("title", data.dataName)
                        startActivity(intent)
                    }
                    "1010303020204","1010303020205","1010303020206","1010303020207","1010303020208","1010303020209" -> {
                        val intent = Intent(mContext, ActivityWebView::class.java)
                        intent.putExtra("title", data.dataName)
                        intent.putExtra("url", data.url)
                        intent.putExtra("shareContent", data.dataName)
                        startActivity(intent)
                    }
                    else -> {//农业气象
                        if (data.childList.size > 0) {
                            val intent = Intent(mContext, ActivityServerHyqx::class.java)
                            intent.putExtra("title", data.dataName)
                            val bundle = Bundle()
                            bundle.putParcelableArrayList("dataList", data.childList)
                            intent.putExtras(bundle)
                            startActivity(intent)
                        }
                    }
                }
            }

            if (data.dataName != null) {
                holder.itemName!!.text = data.dataName
            }
            holder.itemimageview_top!!.visibility = View.GONE
            return view!!
        }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (resultCode == Activity.RESULT_OK) {
            when (requestCode) {
                CONST.RESULT_LOGIN -> {
                    val intent = Intent(this, ActivityPhotoShow::class.java)
                    intent.putExtra("title", "农情苗情速报")
                    intent.putExtra("imgType", "2")//imgType:图片类型，1（实景开拍），2（农业开拍分类）必须传，区分哪个业务
                    startActivity(intent)
                }
            }
        }
    }

}
